pkg/fileutils/fileutils.go

```diff
@@ -57,63 +57,30 @@ func NewPatternMatcher(patterns []string) (*PatternMatcher, error) {
 	return pm, nil
 }
 
-// Deprecated: Please use the `MatchesResult` method instead.
-// Matches matches path against all the patterns. Matches is not safe to be
-// called concurrently
-func (pm *PatternMatcher) Matches(file string) (bool, error) {
```

Comment:
Despite being deprecated, this can break consumers and is hence a breaking change which in turn would require to bump the major version of c/storage to v2.0.0.  Let's not go down this path and leave the API as is.